presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java

```diff
@@ -1267,30 +1499,91 @@ public PrestoS3OutputStream(
                 long multiPartUploadMinFileSize,
                 long multiPartUploadMinPartSize,
                 PrestoS3AclType aclType,
-                PrestoS3StorageClass s3StorageClass)
+                PrestoS3StorageClass s3StorageClass,
+                boolean disableChecksums,
+                S3AsyncClient injectedAsyncClient)
                 throws IOException
         {
             super(new BufferedOutputStream(Files.newOutputStream(requireNonNull(tempFile, "tempFile is null").toPath())));
 
-            transferManager = TransferManagerBuilder.standard()
-                    .withS3Client(requireNonNull(s3, "s3 is null"))
-                    .withMinimumUploadPartSize(multiPartUploadMinPartSize)
-                    .withMultipartUploadThreshold(multiPartUploadMinFileSize).build();
+            requireNonNull(s3, "s3 is null");
+            this.disableChecksums = disableChecksums;
+
+            // Used injected async client if provided
+            if (injectedAsyncClient != null) {
+                log.debug("Using injected S3AsyncClient for testing");
+                this.s3AsyncClient = injectedAsyncClient;
+                this.transferManager = S3TransferManager.builder()
+                        .s3Client(s3AsyncClient)
+                        .build();
+            }
+            else {
+                // Normal path: create real S3AsyncClient
+                S3AsyncClientBuilder asyncBuilder = S3AsyncClient.builder()
+                        .credentialsProvider(requireNonNull(credentialsProvider, "credentialsProvider is null"))
+                        .region(requireNonNull(region, "region is null"))
+                        .serviceConfiguration(S3Configuration.builder()
+                                .checksumValidationEnabled(!disableChecksums)
+                                .build());
+
```

Comment:
The v1 transfer manager multipart configs like `MinimumUploadPartSize` and `MultipartUploadThreshold` are now set via 
```
S3AsyncClient.builder().multipartConfiguration(...);
```